### PR TITLE
Allow extension authors to set valueSelection in a QuickPick/InputBox

### PR DESCRIPTION
**Pull request by r3m0t**
Similar to #119144 @TylerLeonhardt 

Allows `showQuickPick` and `showInputBox` to be used with `step, totalSteps` so that extension authors don't need to use the more complex `createQuickPick` and `createInputBox`.

Also allow `valueSelection` to be set with `createQuickPick` and `createInputBox`.

Test with this extension- https://github.com/r3m0t/vscode-extension-samples/tree/quickinput/quickinput-sample

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
I'll bring this to the API review but I won't get hopes too high. We intentionally haven't done this in the past is because the `pick` API is suppose to be for one-offs picks while the `createQuickPick` is for longer form quick pick use cases (multi-step).

With this proposal, you'll have to keep track of `totalSteps` outside of the quickpick and pass that into `pick` every single time.

I do acknowledge that the `pick` API is simpler than `createQuickPick` and this PR enables:

```ts

const a = await window.showInputBox({
		title: "Example",
		prompt: "Please select step 1's value",
		step: 1,
		totalSteps: 3,
	}, ['a', 'b', 'c']);

const b = await window.showInputBox({
		title: "Example",
		prompt: "Please select step 2's value",
		step: 2,
		totalSteps: 3,
	}, ['a', 'b', 'c']);
```

but that's actually 2 pieces data that are duplicated: `title` and `totalSteps`... which isn't ideal and would be avoided using `createQuickPick`.

**Comment by r3m0t** [comment]
I'm more interested in valueSelection actually,  which I implemented in this PR. I am using the multi step input and want to set valueSelection on the second step. I just added step to QuickPickOptions for symmetry.

**Comment by TylerLeonhardt** [comment]
Yes that's right, I remember your issue. I'd rather have `valueSelection?: [number, number]` on `InputBox` then `totalSteps` on `pick`.

**Comment by TylerLeonhardt** [comment]
Hi @r3m0t we discussed this in our API review because I wanted to get you an answer for taking this on.

We all feel similar to what I outlined above with the steps, but would 💯 take a PR with the `valueSelection`. If you update this PR with that, I'd happily get that in!

**Comment by r3m0t** [comment]
@TylerLeonhardt there you go, I kept the other branch in `r3m0t/quick-pick-steps-c`

**Review by TylerLeonhardt — APPROVED** [review]

File: src/vscode-dts/vscode.d.ts [hunk 1]
```diff
@@ -2018,10 +2018,10 @@ declare module 'vscode' {
 		/**
 		 * Selection of the pre-filled {@linkcode InputBoxOptions.value value}. Defined as tuple of two number where the
 		 * first is the inclusive start index and the second the exclusive end index. When `undefined` the whole
-		 * word will be selected, when empty (start equals end) only the cursor will be set,
+		 * pre-filled value will be selected, when empty (start equals end) only the cursor will be set,
 		 * otherwise the defined range will be selected.
 		 */
-		valueSelection?: [number, number];
+		valueSelection?: readonly [number, number];
```
Comment (TylerLeonhardt) [inline src/vscode-dts/vscode.d.ts:2024]: This is a slightly breaking change so please leave this as it is for now.

**Review by TylerLeonhardt — CHANGES_REQUESTED** [review]

**Comment by r3m0t** [comment]
@TylerLeonhardt all sorted

**Review by TylerLeonhardt — DISMISSED** [review]

**Review by joaomoreno — DISMISSED** [review]

File: src/vs/workbench/browser/actions/workspaceCommands.ts [hunk 2]
```diff
@@ -235,7 +235,7 @@ CommandsRegistry.registerCommand({
 		return commandService.executeCommand('_files.newWindow', commandOptions);
 	},
 	description: {
-		description: 'Opens an new window depending on the newWindow argument.',
+		description: 'Opens an new window depending on the reuseWindow argument.',
```
Comment (jrieken) [inline src/vs/workbench/browser/actions/workspaceCommands.ts:238]: unrelated change sin this file?

Comment (r3m0t) [reply src/vs/workbench/browser/actions/workspaceCommands.ts:238]: It is an accurate change to the description, but you can remove it if you like

Comment (r3m0t) [reply src/vs/workbench/browser/actions/workspaceCommands.ts:238]: @jrieken removed

Comment (TylerLeonhardt) [reply src/vs/workbench/browser/actions/workspaceCommands.ts]: In this case, it's ok. I've already reviewed all the changes but @jrieken brings up a really good point for future PRs.

**Review by TylerLeonhardt — APPROVED** [review]

**Review by rzhao271 — APPROVED** [review]
